distributed/worker.py

```diff
@@ -2825,15 +2829,16 @@ def get_call_stack(self, comm=None, keys=None):
         result = {k: profile.call_stack(frame) for k, frame in frames.items()}
         return result
 
-    def _notify_transition(self, key, start, finish, **kwargs):
+    def _notify_plugins(self, method_name, *args, **kwargs):
         for name, plugin in self.plugins.items():
-            if hasattr(plugin, "transition"):
-                try:
-                    plugin.transition(key, start, finish, **kwargs)
-                except Exception:
-                    logger.info(
-                        "Plugin '%s' failed with exception" % name, exc_info=True
-                    )
+            try:
+                plugin_method = getattr(plugin, method_name)
+            except AttributeError:
+                continue
```

Comment:
I have a slight preference for `if hasattr(plugin, method_name)` over a try-except.  It seems more clear, and it's also marginally faster when we expect many misses (try-except is faster than if-else if the try block succeeds, but about 5x slower if the except path is taken).